policy-controller/k8s/status/src/index.rs

```diff
@@ -325,6 +320,39 @@ impl kubert::index::IndexNamespacedResource<k8s::policy::Server> for Index {
     // to handle resets specially.
 }
 
+impl kubert::index::IndexNamespacedResource<k8s::Service> for Index {
+    fn apply(&mut self, resource: k8s::Service) {
+        let namespace = resource.namespace().expect("Service must have a namespace");
+        let name = resource.name_unchecked();
+        let id = ResourceId::new(namespace, name);
+
+        self.services.insert(id);
+
+        // If we're not the leader, skip reconciling the cluster.
+        if !self.claims.borrow().is_current_for(&self.name) {
+            tracing::debug!(%self.name, "Lease non-holder skipping controller update");
+            return;
+        }
+        self.reconcile();
+    }
+
+    fn delete(&mut self, namespace: String, name: String) {
+        let id = ResourceId::new(namespace, name);
+
+        self.services.remove(&id);
+
+        // If we're not the leader, skip reconciling the cluster.
+        if !self.claims.borrow().is_current_for(&self.name) {
+            tracing::debug!(%self.name, "Lease non-holder skipping controller update");
+            return;
+        }
+        self.reconcile();
```

Comment:
nit, TIOLI: starting to wonder if we ought to have an `Index::reconcile_if_leader` or something that just does this...